paymentsheet/src/main/java/com/stripe/android/paymentsheet/verticalmode/PaymentMethodRowButton.kt

```diff
@@ -96,5 +226,90 @@ private fun TitleContent(title: String, subtitle: String?, isEnabled: Boolean, c
                 color = if (isEnabled) subtitleTextColor else subtitleTextColor.copy(alpha = 0.6f),
             )
         }
+        if (trailingContent != null) {
+            Row {
+                trailingContent()
+            }
+        }
+    }
+}
+
+@Composable
+private fun PaymentMethodContentExtras(
+    title: String,
+    subtitle: String?,
+    isEnabled: Boolean,
+    contentDescription: String?,
+    isSelected: Boolean,
+    iconContent: @Composable RowScope.() -> Unit,
+    trailingContent: (@Composable RowScope.() -> Unit)? = null,
+    isCheckmarkRow: Boolean = false
+) {
+    Row(
+        horizontalArrangement = Arrangement.spacedBy(12.dp),
+        verticalAlignment = if (isCheckmarkRow && trailingContent != null) {
+            Alignment.Top
+        } else {
+            Alignment.CenterVertically
+        },
+    ) {
+        iconContent()
+        if (isCheckmarkRow) {
+            TitleContent(
+                title = title,
+                subtitle = subtitle,
+                isEnabled = isEnabled,
+                contentDescription = contentDescription,
+                trailingContent = trailingContent
+            )
+
+            if (isSelected) {
+                Spacer(modifier = Modifier.weight(1f))
+                Icon(
+                    imageVector = Icons.Filled.Check,
+                    contentDescription = null,
+                    modifier = Modifier.align(Alignment.CenterVertically),
+                    tint = StripeTheme.getColors(isSystemInDarkTheme()).materialColors.primary
+                )
+            }
+        } else {
+            TitleContent(
+                title = title,
+                subtitle = subtitle,
+                isEnabled = isEnabled,
+                contentDescription = contentDescription,
+            )
+
+            if (trailingContent != null) {
+                Spacer(modifier = Modifier.weight(1f))
+                trailingContent()
+            }
+        }
+    }
+}
+
+private fun String?.paddingValues(): Dp = if (this != null) 8.dp else 12.dp
```

Comment:
This is a little deceptive. Maybe a rename to `paddingValuesForSubtitle` would be better? (there's probably still a better name yet).